### ISSUE TITLE
Option to hide or show specific chunks by default

### ISSUE THREAD

**Issue by JulienGAMartin** [bblodfon/rtemps#1]
Hi
for the bookdown_lite, it seems that all the code chunks are either hidden or shown by default (when changing  `window.initializeCodeFolding("show" === "hide")`. Is it possible to show all chunks except for a few? 
So if by default all chunks are hidden can I specify an option to show a specific chunk by default similarly to the `class.source = 'fold-show'` option in html_document ?

**Comment by bblodfon** [comment]
Hi!

The behavior of the top-right button is binary, it either shows them all or collapses them all as you've seen. What you are asking me is now supported (I hacked it in)! Install the github version and it should work on any R block annotated with the `class.source = 'fold-show'` option.

**Comment by JulienGAMartin** [comment]
Sorry not working properly
using  the option `class.source = 'fold-show'` in any of the R block will show the code for the block and all following blocks on the page .
So adding the option to the second block in the example lead to the first block being hidden and all other blocks being shown by default.

**Comment by bblodfon** [comment]
Hey! Thanks for looking it, small bug on my hack :) It should work now!

**Comment by JulienGAMartin** [comment]
Yes it is working well now. Thanks.
I made a small change to the code so now it can respond to both `fold-show` and `fold-hide` classes to show or hide by default.
The only thing missing now would be an easy way to set if by default all code should be shown or hidden. 
Right now I have to change by hand `window.initializeCodeFolding("show" === "hide");` to hide or show all code at init.

**Comment by bblodfon** [comment]
As you've already have seen, this is a very 'hacked' solution and you can do whatever you want with it.

My motivation for having all code blocks hidden at init, was that the average user won't bother to look at the code when given an analysis report with text, text-boxes, code and figures (mostly they will focus on the figures and boxes, the eye-catching stuff). Adding the functionality to show some code blocks at init was a nice feature. But showing them all at init goes against my core design/philoshophy for this meta-package!

**Comment by JulienGAMartin** [comment]
Thanks. I understand your design/philosophy. I was just asking and also interested in you 'hack' because the option `code_folding` option that work for html_book is not supported in `gitbook` format. Your 'hack' is solving this problem and the only missing feature to fully provide an equivalent to the `code_folding` option   is an easy way to choose if the code chunks are shown or hidden at the beginning. I can see some people typing tutorials and wanting all code shown except for a few section of exercises for example.
I am not asking you to implement it, I am just giving my opinion and why I asked for it originally.

**Comment by AndyBunn** [comment]
Hello and thanks for the `cold_folding` option to `gitbook`. I too would like to init with all the code on and hide chunks as needed for exactly the reason mentioned by above. I have tutorials and normally show all the code and hide it only when there is some good reason (e.g., complex plotting). I have never worked js so I'm unsure of what to change in `hide_code.html` but gather that is is something in `window.initializeCodeFolding("show" === "hide");` Any tips appreciated. And thanks again.

EDIT: I figured it out. For those coming after, I changed `window.initializeCodeFolding("show" === "hide");` to `window.initializeCodeFolding("show" !== "hide");` to show all the code on init.
